### ISSUE TITLE
Can I regenerate certs after changing the root domain?

### ISSUE THREAD

**Issue by bwbb** [caprover/caprover#695]
I've just tested my caprover setup on a personal domain, everything looks good, and now I'm ready to setup the real domain. All the installed apps are already ready for production. I've changed my root domain via the CapRover UI, and regular HTTP traffic works fine.  Is it possible to re-generate the SSL certificates with the new hostname?

If not, what's the easiest way to get SSL back - perferrably without setting up the apps again?
- Backup/Restore?
- or do I need to reinstall from scratch?

**Comment by githubsaturn** [comment]
When you change the root domain the SSL certs for the new domain should be automatically recreated (unless there is failure in the flow).

Also, keep in mind that the root domain doesn't need to be seen by your user. In fact, it's better to hide it behind an obsecure domain like `captain.g928hg72h.server.yourdomain.com`. You can then connect your real domain by navigating to Apps > Your App > Connect New domain for each app:
![image](https://user-images.githubusercontent.com/10582082/82971723-628a4000-9fa1-11ea-9cb8-d388de1f9e61.png)

Does this answer your question?

**Comment by bwbb** [comment]
>In fact, it's better to hide it behind an obsecure domain like `captain.g928hg72h.server.yourdomain.com`. 

This ^ is a really cool feature! I will try that...

But for now my root domain is now `apps.MyDomain.com`, with my captain url at `captain.apps.mydomain.com`. When I connect a new domain for a particular app as you described above, I set it up as `myapp.mydomain.com`, and at that url, the HTTPS link works. However, if I try to access it via the default/automatic url `https://myapp.apps.mydomain.com`, Chrome gives a not-secure error: `ERR_CERT_INVALID`... Also, the captain URL is not accessible via HTTPS on the new domain. Am I missing something?

So basically, is there any way to get the official/automatic (both the dashboard, and app-specific) URLs working again via HTTPS after the domain change?

**Comment by githubsaturn** [comment]
Once you update your root domain through dashboard, CapRover tries to automatically recreate the SSL certs for domains that had been attached to an SSL certificate. This process may fail (such as network issues). In that case, you'll be left with a non-https domain.

But regardless, you should be able to click on enable HTTPS next to your domain. Is that button disabled for you? Like this:
![image](https://user-images.githubusercontent.com/10582082/83077539-9cad1d80-a045-11ea-82a6-fae1b636cab0.png)

If no, simply click on it to enable the HTTPS. There is no automatic process for this. If it's not enabled, it means that something during the domain change interrupted the config change. If this is the case, you'll have to manually edit your config file:
```
docker service scale captain-captain=0
nano /captain/data/config-captain.json
# make adjustment to enable SSL flags, save and exit
docker service scale captain-captain=1
```

**Comment by bwbb** [comment]
Maybe it just took a little while... It's all working now - Thanks!
